modules/highgui/src/write_mat_to_xrgb8888.simd.hpp

```diff
@@ -0,0 +1,129 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include "opencv2/core/hal/intrin.hpp"
+
+
+namespace cv {
+CV_CPU_OPTIMIZATION_NAMESPACE_BEGIN
+
+void write_mat_to_xrgb8888(cv::Mat const &img, void *data);
```

Comment:
Please avoid using Mat in optimized dispatched code. Use void*, step, size instead.

Reply:
Thank you for your review, I fixed it.

```
Wayland Backend
|
write_mat_to_xrgb8888(const cv::Mat& img, void *data)
  + cv::opt_SSE4_1::write_g8_to_xrgb8888(uint8_t* src, uint8_t* dst, int img_col);
  + cv::opt_SSE4_1::write_bgr888_to_xrgb8888(uint8_t* src, uint8_t* dst, int img_col);
  + cv::opt_SSE4_1::write_bgra8888_to_xrgb8888(uint8_t* src, uint8_t* dst, int img_col);
  + cv::opt_AVX2::write_g8_to_xrgb8888(uint8_t* src, uint8_t* dst, int img_col);
  + cv::opt_AVX2::write_bgr888_to_xrgb8888(uint8_t* src, uint8_t* dst, int img_col);
  + cv::opt_AVX2::write_bgra8888_to_xrgb8888(uint8_t* src, uint8_t* dst, int img_col);
```